server/sockets.js

```diff
@@ -567,31 +590,14 @@ if (cluster.isMaster) {
 			room = rooms.get(roomid);
 			if (!room) return;
 
-			/** @type {[string?, string?, string?]} */
-			let messages = [null, null, null];
+			/** @type {[string?, string?, string?, string?, string?]} */
+			const messages = [null, null, null, null, null];
 			message = data.substr(nlLoc + 1);
 			roomChannel = roomChannels.get(roomid);
 			for (const [socketid, socket] of room) {
-				switch (roomChannel ? roomChannel.get(socketid) : '0') {
-				case '1':
-					if (!messages[1]) {
-						messages[1] = message.replace(/\n\|split\n[^\n]*\n([^\n]*)\n[^\n]*\n[^\n]*/g, '\n$1').replace(/\n\n/g, '\n');
-					}
-					socket.write(messages[1]);
-					break;
-				case '2':
-					if (!messages[2]) {
-						messages[2] = message.replace(/\n\|split\n[^\n]*\n[^\n]*\n([^\n]*)\n[^\n]*/g, '\n$1').replace(/\n\n/g, '\n');
-					}
-					socket.write(messages[2]);
-					break;
-				default:
-					if (!messages[0]) {
-						messages[0] = message.replace(/\n\|split\n([^\n]*)\n[^\n]*\n[^\n]*\n[^\n]*/g, '\n$1').replace(/\n\n/g, '\n');
-					}
-					socket.write(messages[0]);
-					break;
-				}
+				const channelid = roomChannel ? Number(roomChannel.get(socketid)) : 0;
```

Comment:
So `channelid` is guaranteed to be an integer from 0 to 4? That should probably be documented somewhere.

Reply:
In fact, `type ChannelID = 0 | 1 | 2 | 3 | 4` would probably help.

Reply:
```javascript
/**
 * roomid:socketid:channelIndex
 * @type {Map<string, Map<string, string>>}
 */
const roomChannels = new Map();
```

No earthly close why `channelIndex` is a `string`, but the fact that its an `index` means I'm now more confident its not going to be negative.

Reply:
Ah, its a `string` because it comes from `'.roomid, .channelid, .socketid'` which are all strings.

Reply:
Yep, I've investigated at it's in fact `0 | 1 | 2 | 3 | 4`.

Now all that remains is documenting it so people don't accidentally pass the wrong thing in.

Reply:
Added a `ChannelID` type. Converting `roomid`/`socketid` to be numbers is I think out of scope.